### ISSUE TITLE
Data update in equipment endpoints

### ISSUE THREAD

**Issue by sadullahgultekin** [bounswe/bounswe2019group6#191]
Hi fellows,

Currently I am implementing the portfolio page. I want to show the user all trading equipments and let the user select the ones that he/she wants to add to the related portfolio. To be able to retrieve this information, I use `/equipment/currency/list`, `/equipment/crypto-currency/list` and `/equipment/stock/list` endpoints. These are returning data like:
```
{
     equipments: [
     "JPY",
     "EUR",
     "TRY",
     "CZK",
     "GBP",
     "CNY"
     ],
     base: "USD"
}
```
However, this is not enough for me to show the user. The user should be selecting which equipment to add the system by looking their current value or increase/decrease rate etc. 

One solution is to send a request to the backend for each equipment separately, because `/equipment/{name}` endpoint gives me the desired information. But this solution would create an overhead to the system, so I want to avoid it. 

Another solution that comes to my mind is we can update the three endpoints mentioned above so that they have enough data. For example we can have a response like:
```
{ 
   equipments:{ 
      "JPY": { 
         "currentStock":0,
         "currentValue":0
      },
      "EUR": { 
         "currentStock":0,
         "currentValue":0
      },
      "TRY": { 
         "currentStock":0,
         "currentValue":0
      }
   },
   base:"USD"
}
```
What do you think? Should we update the endpoints like above?

**Comment by enozcan** [comment]
Hi @sadullahgultekin 

Seems reasonable. What android team thinks about that? @bwqr @mstfalp @barandenizkorkmaz 

If the format you mentioned also OK for the android team, we can modify it accordingly.

**Comment by irmakguzey** [comment]
Hi @sadullahgultekin ,

With the format you mentioned, I believe there is something missing. If we pull 
`
{ 
   equipments:{ 
      "JPY": { 
         "currentStock":0,
         "currentValue":0
      },
      "EUR": { 
         "currentStock":0,
         "currentValue":0
      },
      "TRY": { 
         "currentStock":0,
         "currentValue":0
      }
   },
   base:"USD"
}
`

this as the `equipment/currency/list` respond, since we don't have the keys of each equipment in the beginning we couldn't receive the datas you want. So the format, I believe, should be something like this: 

`
{ 
   equipments: [
      {key: "JPY",
       data: { 
         "currentStock":0,
         "currentValue":0
        }
      },
      {key: "EUR",
       data: { 
         "currentStock":0,
         "currentValue":0
        }
      },
      {key: "TRY",
       data: { 
         "currentStock":0,
         "currentValue":0
        }
      },
   ],
   base:"USD"
}
`

**Comment by sadullahgultekin** [comment]
Thank you @irmakguzey, I didn't know that we needed a key, we can add such field.

What I wanted to emphasize though, we need more informative response data in these endpoints. Any data that is thought useful can be added.

**Comment by bwqr** [comment]
I think we should not restrict the incoming data. For example, if we look into the `equipment/{name}` endpoint, we can see lots of information. These information should also be included in the listing endpoints. One more thing, instead of requesting three different endpoints for one page, we can send a request to endpoint which returns all of the three different equipment types like 
```json
"currency": [],
"crypto_currency": [],
"stock": []
```

**Comment by sadullahgultekin** [comment]
@bwqr Wouldn't it be overhead for the system? There are some pages like portfolio creation page  that we want to list general information about a trading equipment instead of detailed information. 

If you need such endpoint, we can ask to our backend team to implement it. But for web team, limited data would just do the job.

**Comment by enozcan** [comment]
Hi @sadullahgultekin @bwqr @irmakguzey 
Can you make a consensus concerning this?

**Comment by sadullahgultekin** [comment]
We are in a hurry, we need to implement new features in this week. I think ırmak's suggestion solves most of the problems. We can do it like that.

**Comment by enozcan** [comment]
The updated version is the following as @irmakguzey suggested:

```json
{"equipments":[{"code":"JPY","data":{"currentValue":108.63,"currentStock":1000000.0}},
{"code":"EUR","data":{"currentValue":0.9074,"currentStock":1000000.0}},
{"code":"TRY","data":{"currentValue":5.7135,"currentStock":1000000.0}}],"base":"USD"}
```

Fixed on 85f0fd3535ca2ed44eddb03ed015fc167941dc79 and will be deployed soon.
Thanks for the feedback.
